python-avd/pyavd/schema/generate_docs/tablerowgen.py

```diff
@@ -0,0 +1,355 @@
+        """
+        Render markdown for "required" field.
+        """
+        if self.schema._is_primary_key:
+            return "Required, Unique"
+        if self.schema.required:
+            return "Required"
+
+    def render_default(self) -> str | None:
+        """
+        Should render markdown for "default" field.
+        """
+        if self.schema.default is not None:
+            if isinstance(self.schema.default, (list, dict)) and len(self.schema.default) > 1:
+                return "See (+) on YAML tab"
+
+            return f"`{self.schema.default}`"
+
+    def render_description(self) -> str | None:
+        """
+        Renders markdown for "description" field including deprecation text with color.
+        """
+        descriptions = []
+        if self.schema.description:
+            descriptions.append(self.schema.description.replace("\n", "<br>"))
+        description = " ".join(descriptions)
+
+        if deprecation := self.get_deprecation_description():
+            description += deprecation
+
+        return description or None
+
+    def render_children(self) -> Generator[TableRow]:
+        """Noop for classes without children. Override in subclasses for dict and list."""
+        yield from []
+
+    def render_restrictions(self) -> str | None:
+        """
+        Renders markdown for "restrictions" field as a multiline text compatible with a markdown table cell.
+
+        Only covers generic restrictions. Should be overridden in type specific subclasses.
+        """
+        restrictions = []
+        valid_values = []
+        if getattr(self.schema, "dynamic_valid_values", None):
+            valid_values.append(f"<value(s) of {self.schema.dynamic_valid_values}>")
+
+        if getattr(self.schema, "valid_values", None):
+            valid_values.extend(self.schema.valid_values)
+
+        if valid_values:
+            restrictions.append("Valid Values:")
+            # TODO: Remove legacy output
+            if LEGACY_OUTPUT:
+                restrictions.extend(f"- {valid_value}" for valid_value in valid_values)
+            else:
+                restrictions.extend(f"- <code>{valid_value}</code>" for valid_value in valid_values)
+
+        return "<br>".join(restrictions) or None
+
+
+class TableRowGenBool(TableRowGenBase):
+    pass
+
+
+class TableRowGenInt(TableRowGenBase):
+    def render_restrictions(self) -> str | None:
+        """
+        Renders markdown for "restrictions" field as a multiline text compatible with a markdown table cell.
+
+        Leverages common restrictions from base class.
+        """
+        restrictions = []
+        if self.schema.min is not None:
+            restrictions.append(f"Min: {self.schema.min}")
+        if self.schema.max is not None:
+            restrictions.append(f"Max: {self.schema.max}")
+
+        if base_restrictions := super().render_restrictions():
+            restrictions.extend(base_restrictions.split("<br>"))
```

Comment:
we could probably avoid spliting to rejoin everything just after

something like

```
class TableRowGenBase:
  def render_restrictions(self) -> str | None:
    return "<br>".join(self._get_restrictions_list) or None

    def _render_restrictions(self) -> list[str]:
        """ This is what you should override in child classes"
        restrictions = []
        valid_values = []
        [...]
       
        return restrictions

[...]

class TableRowGenInt(TableRowGenBase):
    def _render_restrictions(self) -> list[str]:
        restrictions = []
        if self.schema.min is not None:
            restrictions.append(f"Min: {self.schema.min}")
        if self.schema.max is not None:
            restrictions.append(f"Max: {self.schema.max}")
        restrictions.extend(super()._render_restrictions()

        return restrictions
```

Reply:
The nested function would not work well with subclasses so I will split this into two - a method and a @property to do this in a cleaner way.